### PR TITLE
feat(perf): Improve experimental performance facets api

### PR DESCRIPTION
**Pull request by k-fish**
### Summary 
This brings back sorting tag keys by count, which also allows sampling to only happen at a minimum number of tag values. In addition it adds another aggregation to allow sorting by count and adds a new named tuple instead of mis-using the existing one.

### CODE REVIEW

File: src/sentry/snuba/discover.py [hunk 1]
```diff
@@ -777,12 +780,12 @@ def get_performance_facets(
     with sentry_sdk.start_span(op="discover.discover", description="facets.frequent_tags"):
         # Get the tag keys with the highest deviation
```
Comment (wmak) [inline src/sentry/snuba/discover.py:781]:
```suggestion
        # Get the most relevant tag keys
```

File: src/sentry/snuba/discover.py [hunk 2]
```diff
@@ -796,8 +799,10 @@ def get_performance_facets(
             return []
 
     results = []
+    snuba_filter.conditions.append([aggregate_column, "IS NOT NULL", None])
 
-    sampling_enabled = True
+    # Only enable sampling if over 10000 values
```
Comment (wmak) [inline src/sentry/snuba/discover.py:804]: out of curiosity was 10k chosen arbitrarily?

**Review by wmak — APPROVED** [review]
smol comment suggestion, and a question about a number otherwise LGTM

File: src/sentry/snuba/discover.py [hunk 3]
```diff
@@ -796,8 +799,10 @@ def get_performance_facets(
...
+    sampling_enabled = True if (key_names["data"][0]["count"] > 10000) else False
```
Comment (dashed) [inline src/sentry/snuba/discover.py:805]:
```suggestion
    sampling_enabled = key_names["data"][0]["count"] > 10000
```

**Review by dashed — APPROVED** [review]

File: src/sentry/snuba/discover.py (hunk 2 above)
Comment (k-fish) [reply src/sentry/snuba/discover.py:804]: Same as above, presumably yes. Ideally we can set it to hit a target confidence rate, but that can be for later.

File: src/sentry/snuba/discover.py (hunk 3 above)
Comment (k-fish) [reply src/sentry/snuba/discover.py:805]: Lol, derp, thx

File: src/sentry/snuba/discover.py [hunk 4]
```diff
@@ -831,7 +839,9 @@ def get_performance_facets(
             )
             results.extend(
                 [
-                    FacetResult(r["tags_key"], r["tags_value"], int(r["aggregate"]))
+                    PerformanceFacetResult(
+                        r["tags_key"], r["tags_value"], int(r["aggregate"]), int(r["count"])
```
Comment (Zylphrex) [inline src/sentry/snuba/discover.py:843]: Was the aggregate intentionally casted to an `int` here?

```suggestion
                        r["tags_key"], r["tags_value"], float(r["aggregate"]), int(r["count"])
```

**Pull request merged**